src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/VsItemList.cs

```diff
@@ -26,21 +26,45 @@ public VsItemList(IEnumerable<T> collection) : base()
 
         public T Item(Object index)
         {
+            if (index is string)
+            {
+                TryGetValue((string)index, out var value);
+                return value;
+            }
+
             try
             {
-                if (index is string)
-                {
-                    return this[(string)index];
-                }
-                else if (index is int)
+                return this[(int)index];
+            }
+            catch (ArgumentOutOfRangeException)
```

Comment:
It was not clear from the contract whether caller was expecting to handle exceptions. But this really only applies to KeyNotFoundException since callers will likely pass missing string keys. For int indexing, it should be OK to expect a correct index and remove the try-catch.

---
In reply to: [89560607](https://github.com/dotnet/roslyn-project-system/pull/816#discussion_r89560607) [](ancestors = 89560607)